inc/console/migration/buildmissingtimestampscommand.class.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2018 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Console\Migration;
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+use Glpi\Console\AbstractCommand;
+use Log;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class BuildMissingTimestampsCommand extends AbstractCommand {
+
+   protected function configure() {
+      parent::configure();
+
+      $this->setName('glpi:migration:build_missing_timestamps');
+      $this->setDescription(__('Set missing `date_creation` and `date_mod` values using log entries.'));
+      $this->setHidden(true); // Hide this command as it is when migrating from really old GLPI version
+   }
+
+   protected function execute(InputInterface $input, OutputInterface $output) {
+
+      global $CFG_GLPI;
+
+      $types = array_merge(
```

Comment:
I just get existing list (and removed `NetworkEquipmentFirmware` that does not exists anymore). I am not really sure it is up to date.

Maybe we should just loop on all tables (except `glpi_logs`) and update them if fields `date_creation` and/or `date_mod` are found.

Reply:
Done in 4f9bd20b72645d3353cb7a8fe6de5bf77fdd25f4.